src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOIFinder.java

```diff
@@ -0,0 +1,62 @@
+package org.openstreetmap.atlas.geography.atlas.items.complex.aoi;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.items.complex.Finder;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.collections.MultiIterable;
+
+/**
+ * {@link ComplexAOI} finder.
+ *
+ * @author sayas01
+ */
+public class ComplexAOIFinder implements Finder<ComplexAOI>
+{
+    /**
+     * Finds all relations and areas that are candidates for {@link ComplexAOI} and converts them
+     * into {@link ComplexAOI}.
+     *
+     * @param atlas
+     *            The {@link Atlas} to browse.
+     * @return {@link Iterables} of {@link ComplexAOI}.
+     */
+    @Override
+    public Iterable<ComplexAOI> find(final Atlas atlas)
+    {
+        final Iterable<ComplexAOI> iterableOfComplexAOIRelations = StreamSupport
+                .stream(atlas.relations().spliterator(), false).map(ComplexAOI::getComplexAOI)
```

Comment:
Just curious, what if you set the parallel flag to true here, does that have any impact on results or runtime?

Reply:
I tested the run time with sequential and parallel stream for a Test Atlas and found that the run time for parallel stream to be  smaller (by a couple of ms) than the sequential. It would vary with machine and Atlas. For now, with the result I got, I have updated the stream to be parallel.